File: _Enutrof_[__-1]🎧🌝💉☕/Day 63/level-order-traversal.py
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        result = []
        
        if not root:
            return result
        
        level = [root]
        while level:
            temp = []
            values = []
            for node in level:
                values.append(node.val)
                if node.left:
                    temp.append(node.left)
                if node.right:
                    temp.append(node.right)
            level = temp
            result.append(values)
        return result

File: Engr-T.stark/levelOrderTraversal.cpp
queue<TreeNode*> q;
    q.push(root);
    q.push(NULL);
    vector<vector<int>> ans;
    vector<int> t;
    
    
    if(root==NULL)
    {
        return ans;
    }
    
    while(!q.empty())
    {
        TreeNode* temp =q.front();
        q.pop();
        
        if(temp==NULL)
        {   
            ans.push_back(t);
            t.clear();
            if(!q.empty())
            {
                q.push(NULL);
            }
        }
        else  //temp != null
        {
            t.push_back(temp->val);
            
            if(temp->left!=NULL)
            {
                q.push(temp->left);
            }
            if(temp->right!=NULL)
            {
                q.push(temp->right);
            }
        }
    }
    
    return ans;
